daliuge-translator/dlg/dropmake/scheduler.py

```diff
@@ -513,7 +513,7 @@ def __init__(self, drop_list, max_dop=8, dag=None):
         else:
             self._dag = dag
         self._max_dop = max_dop
-        self._parts = None  # partitions
+        self._parts = []  # partitions
```

Comment:
This is similar to changes above, in which we have `None` types being returned but we are treating them like iterators. This makes it clear across the board that we only expect to have a list, which can be empty, rather than having to go to all the places `self._parts` is mentioned and check for `None`.